### ISSUE TITLE
This showed up on my terminal and won't let me access my account from it anymore.

### ISSUE THREAD

**Issue by luisjalbizu** [instabot-py/instabot.py#1390]
Check this or we will delete your issue. (fill in the checkbox with an X like so: [x])
- [x] I have searched for other issues with the same problem. (including #1153)

## Hosting
- [ ] Are you hosting the bot on a VPS or other Cloud hosting?
- [ ] If so; have you tried running the code on your local machine?

#### Select one:
- [ ] Bug
- [ ] Feature Request
- [x] Technical Help
- [ ] Other

### Operating System? (include version)
- [x] MAC
- [ ] Windows
- [ ] Linux
- [ ] Android

## Python Version?
- [ ] 2.x
- [x] 3.x

## Description of your issue
I was fixing my python code so that it could have more comments for it to supply and when I tried to log it back into the bot, this code started to show up and it wouldn't let me access it anymore.
```
Include the contents of the error.log here (if applicable). < REPLACE THIS TEXT
```

## Config

```
Include the contents of your config here (REMOVE YOUR USERNAME AND PASSWORD!) (if applicable). < REPLACE THIS TEXT
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connection.py", line 171, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/util/connection.py", line 56, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 964, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connection.py", line 196, in connect
    conn = self._new_conn()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connection.py", line 180, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x1026abcf8>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/adapters.py", line 445, in send
    timeout=timeout
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/util/retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='ink361.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1026abcf8>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "example.py", line 52, in <module>
    unfollow_whitelist=['example_user_1', 'example_user_2'])
  File "/Users/user/Desktop/bot/........./instabot.py-master/src/instabot.py", line 232, in __init__
    self.login()
  File "/Users/user/Desktop/bot/......./instabot.py-master/src/instabot.py", line 300, in login
    ui = UserInfo()
  File "/Users/user/Desktop/bot/....../instabot.py-master/src/userinfo.py", line 29, in __init__
    self.hello()
  File "/Users/user/Desktop/bot/....../instabot.py-master/src/userinfo.py", line 34, in hello
    main = self.s.get(self.url_list[self.i_a]["main"])
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/sessions.py", line 525, in get
    return self.request('GET', url, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/adapters.py", line 513, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='ink361.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1026abcf8>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
users-imac:instabot.py-master user$

**Comment by Djuki** [comment]
I have the same issue:

```Traceback (most recent call last):
  File "example.py", line 52, in <module>
    unfollow_whitelist=['example_user_1', 'example_user_2'])
  File "/home/vagrant/code/instabot.py/src/instabot.py", line 232, in __init__
    self.login()
  File "/home/vagrant/code/instabot.py/src/instabot.py", line 300, in login
    ui = UserInfo()
  File "/home/vagrant/code/instabot.py/src/userinfo.py", line 29, in __init__
    self.hello()
  File "/home/vagrant/code/instabot.py/src/userinfo.py", line 34, in hello
    main = self.s.get(self.url_list[self.i_a]["main"])
  File "/home/vagrant/.local/lib/python2.7/site-packages/requests/sessions.py", line 521, in get
    return self.request('GET', url, **kwargs)
  File "/home/vagrant/.local/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/vagrant/.local/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/vagrant/.local/lib/python2.7/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='ink361.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f8a6066c290>: Failed to establish a new connection: [Errno -2] Name or service not known',))```

Maybe we are blocked by the Instagram?

**Comment by dimrizo** [comment]
Same issue.

**Comment by Robcis** [comment]
Same problem!

**Comment by luisjalbizu** [comment]
I found out what the issue was. I will leave the links that can help you out within this message. I hope this fixes and helps your problems out as well guys!

// Step #1

// use this link within the developers console to extract the user id from your instagram account
window._sharedData.entry_data.ProfilePage[0].user.id
// use this link within the developers console to extract the user id from your instagram account’s post
window._sharedData.entry_data.ProfilePage[0].user.owner.id
 
or

https://www.otzberg.net/iguserid/


// Step #2

for now I've managed to get around this issue by hard-coding my instagram user id into instabot.py
1. I went to these url’s and got my user id.
https://www.youtube.com/watch?v=8JqdvQKfB3s
https://github.com/instabot-py/instabot.py/issues/1383
https://www.otzberg.net/iguserid/ 
2. In line 301 in instabot.py i replaced this line self.user_id = ui.get_user_id_by_login(self.user_login) with my user id from above self.user_id = 'user id here' 
3. comment line 29 in userinfo.py (or delete it, this should stop the function from calling ink361 api on initialisation) #self.hello() 
save and run bot


// Links Sources for bot and help.

https://www.youtube.com/watch?v=8JqdvQKfB3s
https://github.com/instabot-py/instabot.py/issues/1383
https://www.otzberg.net/iguserid/

**Comment by DarkSuniuM** [comment]
Closed all the issues till 2019-01-09, We updated the repo, Please create a new issue if you need! Thanks
